From 9c7875fc4f9a11bdd22bd8789c4b2ac3ba14c0cb Mon Sep 17 00:00:00 2001
From: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Date: Fri, 16 Apr 2021 13:20:16 +0300
Subject: [PATCH 225/269] tf-a: fix boot flow from eMMC

eMMC device type would not get selected because of a typo in the macro
that is guarding this selection. Also, there was no way of selecting
the eMMC boot flow when building tf-a. Hence, the platform makefile
now checks the presence of S32G_BOOT_FROM_EMMC when executing make.

Issue: ALB-7007
Upstream-Status: Pending 

Signed-off-by: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/nxp/s32g/mmc/s32g_mmc.c | 2 +-
 plat/nxp/s32g/platform.mk       | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/nxp/s32g/mmc/s32g_mmc.c b/drivers/nxp/s32g/mmc/s32g_mmc.c
index 4904d006c..d01c74b18 100644
--- a/drivers/nxp/s32g/mmc/s32g_mmc.c
+++ b/drivers/nxp/s32g/mmc/s32g_mmc.c
@@ -119,7 +119,7 @@
 #define IDENTIFICATION_MODE_FREQUENCY	(400 * 1000)
 #define MMC_FULL_SPEED_MODE_FREQUENCY	(26 * 1000 * 1000)
 
-#ifdef S23G_BOOT_FROM_EMMC
+#ifdef S32G_BOOT_FROM_EMMC
 static struct mmc_device_info device_info = {
 	.mmc_dev_type = MMC_IS_EMMC,
 };
diff --git a/plat/nxp/s32g/platform.mk b/plat/nxp/s32g/platform.mk
index a5c8c9921..3b010220d 100644
--- a/plat/nxp/s32g/platform.mk
+++ b/plat/nxp/s32g/platform.mk
@@ -195,3 +195,7 @@ ifneq ($(BL32_EXTRA1),)
 $(eval $(call TOOL_ADD_IMG,bl32_extra1,--tos-fw-extra1))
 endif
 
+# Check if we are booting from eMMC
+ifneq ($(S32G_BOOT_FROM_EMMC),)
+$(eval $(call add_define,S32G_BOOT_FROM_EMMC))
+endif
-- 
2.17.1

